packages/cli/src/ui/components/shared/vim-buffer-actions.ts

```diff
@@ -674,7 +765,64 @@ export function handleVimAction(
       let offset = getOffsetFromPosition(cursorRow, cursorCol, lines);
 
       for (let i = 0; i < count; i++) {
-        offset = findWordEnd(getText(), offset);
+        const newOffset = findWordEnd(getText(), offset);
+
+        // Special handling for the first iteration when we're at end of word
+        if (i === 0) {
+          const currentLine = lines[cursorRow] || '';
+          const atEndOfWord =
+            cursorCol < currentLine.length &&
+            /\w/.test(currentLine[cursorCol]) &&
```

Comment:
you need to use the codepoint helpers for all this logic to robustly handle unicode characters